tesla_ble_mqtt/libproduct.sh

```diff
@@ -0,0 +1,79 @@
+#
+# Home Assistant Add-On product's library
+#
+function initConfigVariables() {
+
+  ### Required Configuration Settings
+  export BLE_MAC_LIST="$(bashio::config 'ble_mac_list')"
+  export MQTT_SERVER="$(bashio::config 'mqtt_server')"
+  export MQTT_PORT="$(bashio::config 'mqtt_port')"
+  export MQTT_USERNAME="$(bashio::config 'mqtt_username')"
+  export MQTT_PASSWORD="$(bashio::config 'mqtt_password')"
+  export VIN_LIST="$(bashio::config 'vin_list')"
+
+  ### Optional Configuration Settings
+  if bashio::config.exists 'ble_cmd_retry_delay'; then
+    export BLE_CMD_RETRY_DELAY="$(bashio::config 'ble_cmd_retry_delay')"
+  else
+    export BLE_CMD_RETRY_DELAY=""
+  fi
+
+  if bashio::config.exists 'debug'; then
+    DEBUG="$(bashio::config 'debug')"
+    if [ $DEBUG == "true" ]; then
+      bashio::log.level debug
+    fi
+  else
+    export DEBUG=""
+  fi
+
+  if bashio::config.exists 'presence_detection_ttl'; then
+    export PRESENCE_DETECTION_TTL="$(bashio::config 'presence_detection_ttl')"
+  else
+    export PRESENCE_DETECTION_TTL=""
+  fi
+
+  if bashio::config.exists 'presence_detection_loop_delay'; then
+    export PRESENCE_DETECTION_LOOP_DELAY="$(bashio::config 'presence_detection_loop_delay')"
+  else
+    export PRESENCE_DETECTION_LOOP_DELAY=""
+  fi
+
+  # Prevent bashio to complain for "unbound variable"
+  export BLE_LN_LIST=""
+  export BLECTL_FILE_INPUT=""
+  export COLOR=true
+  export ENABLE_HA_FEATURES=""
+  export PRESENCE_EXPIRE_TIME_LIST=""
+
+}
+
+#
+# Definition functions to call bashio::log
+#
+function log_debug() {
+  bashio::log.debug "$1"
```

Comment:
I wonder if we would need $@ instead of $1?

Reply:
$1 is enough cuz everything coming here is wrapped with " " or ' ' - it's a message. Personally I've never seen a log not properly wrapped to make it 1 argument.